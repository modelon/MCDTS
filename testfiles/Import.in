P2.M1
P2.M2
P2.M3
P2.M4
P2.M5
P2.M6
P2.M7
P2.M8
