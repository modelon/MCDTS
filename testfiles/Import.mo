package P

    package R
        constant Real k = 0;
        
        model A
        end A;
    end R;

end P;

package P2
    
    model M1 // Tests class access with qualified import
        import P.R;
        R.A a;
    end M1;
    
    model M2 // Tests component access with qualified import
        import P.R;
        Real x = R.k;
    end M2;
    
    model M3 // Tests qualified access to component in class imported with unqualified import
        import P.*;
        Real x = R.k;
    end M3;
    
    model M4 // Tests access to component imported with unqualified import
        import P.R.*;
        Real x = k;
    end M4;
    
    model M5 // Tests access to class imported with unqualified import
        import P.R.*;
        A a;
    end M5;
    
    model M6 // Tests unqualified import
        import P.R.*;
        A a;
    end M6;
    
    model M7 // Tests named import
        import B = P.R.A;
        B a;
    end M7;
    
    model M8 // Tests class access with named import
        import B = P.R;
        B.A a;
    end M8;
    
end P2;
