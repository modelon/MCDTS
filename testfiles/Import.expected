P, P.R, P.R.A
P, P.R
P, P.R
P, P.R
P, P.R, P.R.A
P, P.R, P.R.A
P, P.R, P.R.A
P, P.R, P.R.A
